react-app/src/components/templates/MovieList/MovieListWrapperComponent.tsx

```diff
@@ -1,14 +1,21 @@
-import React, { FunctionComponent, useEffect, useState } from 'react';
-import { useSelector, useDispatch } from "react-redux";
+import React, { FunctionComponent, useEffect, Suspense } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useParams } from 'react-router-dom';
 
-import { IMovie } from '../../molecules/MovieCard/IMovie';
 import { IMenuItem } from '../../molecules/Navigation/INavigation';
-import NavigationComponent from '../../molecules/Navigation/NavigationComponent';
-import MovieListComponent from '../../organisms/MovieList/MovieListComponent';
-import { MainWrapper, MovieCountWrapper } from './MovieListWrappers';
+import { MainWrapper, MovieCountWrapper, NoMovieFound } from './MovieListWrappers';
 import LoaderComponent from '../../atoms/Loader/LoaderComponent';
-import { actionApiFilterRequest } from '../../../actions/Actions';
+import {
+  actionApiFilterRequest,
+  actionSetFilter,
+  actionSelectMovieId,
+  actionControlVisibility,
+} from '../../../actions/Actions';
+import NavigationComponent from '../../molecules/Navigation/NavigationComponent';
 
+const HeaderComponent = React.lazy(() => import('../../organisms/Header/HeaderComponent'));
+const ErrorBoundary = React.lazy(() => import('../../atoms/ErrorBoundary/ErrorBoundary'));
```

Comment:
I don't think it is a good idea to lazy-load `<ErrorBoundary />` component. Just think about it - what if some error will be thrown before it is loaded?

Reply:
React.lazy removed as it is not supported with SSR.